res/controllers/Denon-MC7000-scripts.js

```diff
@@ -540,23 +549,66 @@ MC7000.PadButtons = function(channel, control, value, status, group) {
     }
 };
 
+// Toggle Shift Button
+MC7000.shiftButton = function(channel, control, value, status, group) {
+    var deckNumber = script.deckFromGroup(group);
+    MC7000.shift[deckNumber - 1] = ! MC7000.shift[deckNumber - 1];
+    midi.sendShortMsg(0x90 + deckNumber - 1, 0x32,
+        MC7000.shift[deckNumber - 1] ? 0x7F : 0x01);
+};
+
 // Toggle Vinyl Mode
 MC7000.vinylModeToggle = function(channel, control, value, status, group) {
-    if (value === 0x00)
+    if (value === 0x00) {
         return; // don't respond to note off messages
-
+    }
     if (value === 0x7F) {
         var deckNumber = script.deckFromGroup(group);
-        MC7000.isVinylMode[deckNumber] = !MC7000.isVinylMode[deckNumber];
-        midi.sendShortMsg(0x90 + channel, 0x07,
-            MC7000.isVinylMode[deckNumber] ? 0x7F : 0x01);
+        MC7000.isVinylMode[deckNumber - 1] = !MC7000.isVinylMode[deckNumber - 1];
+        midi.sendShortMsg(0x90 + deckNumber - 1, 0x07,
+            MC7000.isVinylMode[deckNumber - 1] ? 0x7F : 0x01);
```

Comment:
The same goes for the other functions in this file.

Reply:
that is a valid improvement. I check for other functions and apply:
```
if (value === 0) {
} else {
}
```
instead of checking value again for 0x7F. sending another commit soon.

Reply:
If you return in the 0-case, there's no need for an else case.

Reply:
unnecessary conditions are deleted. Hope I did not overlooked something. Thanks again